packages/stardog-language-utils/src/AbstractLanguageServer.ts

```diff
@@ -148,13 +155,17 @@ ${currentRule}
 
   parseDocument(document: lsp.TextDocument) {
     const content = document.getText();
-    const { cst, errors } = this.parser.parse(content);
+    const { cst, errors, ...otherParseData } = this.parser.parse(content);
```

Comment:
It looks like we're only using this for `semanticErrors` right now. Do we expect this to have any other use in the future? I think it's a bit more readable to deconstruct it explicitly. I'm mostly thinking this because it is also returned for the `TurtleParser` and `SrsParser`, in which case it may be a good idea to extend the type definition of `IStardogParser.parse` to have this as an optional property too.

Reply:
I'm basically not sure. My reasons for doing it this way were basically these:

1. There are different types of errors potentially returned from the parsers, and some of them are general whereas others are language-specific (e.g., unknown namespace errors vs. errors that have to do with violations of one particular grammar). It would probably be useful to have them on different properties at some point. But at the same time, there may also be a number of additional properties (both general and specific) added in the future, possibly for supporting operations like "go to definition" and so on. I don't feel _too_ strongly here, but I'd kind of like to avoid "bloat" on the number of properties we add directly to the object returned by the core `parse` method (I don't want it to become a big grab bag), and I'd also like to not have to come back and change it later (if the number of properties _do_ get out of hand), because that would then break the public facing API.
2. I didn't want to have to make a change to the `IStardogParser` interface in `millan` due to a change in this package, and I also like the typings slightly better this way, since, if we modify the `IStardogParser` interface, our typings for other parsers will tell people that there's an optional property that they might (mistakenly) check for on other parsers, when we actually know for a fact that it will never be there (e.g., on the `BaseStardogParser`, which implements `IStardogParser` but never has anything other than `cst` and `errors` on the `parse` result).
3. It makes some sense to me to have the `parse` method return only things that are strictly produced by the core chevrotain parse (i.e., `cst` and `errors`), and then to put any additional things that are custom properties (e..g, things like `semanticErrors` produced by custom visitors) on an "other" property, as here.